web-admin/src/features/organizations/users/AddUsersDialog.svelte

```diff
@@ -98,37 +94,53 @@
       SPA: true,
       validators: schema,
       async onUpdate({ form }) {
+        // Reset failed invites
+        failedInvites = [];
+
         if (!form.valid) return;
         const values = form.data;
         const emails = values.emails.map((e) => e.trim()).filter(Boolean);
         if (emails.length === 0) return;
 
         const succeeded = [];
-        let errored = false;
-        await Promise.all(
-          emails.map(async (email) => {
-            try {
-              await handleCreate(email, values.role, isSuperUser);
-              succeeded.push(email);
-            } catch (error) {
-              console.error("Error adding user to organization", error);
-              errored = true;
-            }
-          }),
-        );
-
-        eventBus.emit("notification", {
-          type: "success",
-          message: `Invited ${succeeded.length} ${succeeded.length === 1 ? "person" : "people"} as ${values.role}`,
-        });
-
-        if (errored) {
+        const failed = [];
+
+        // Process invites sequentially to maintain order
+        for (const email of emails) {
+          try {
+            await handleCreate(email, values.role, isSuperUser);
+            succeeded.push(email);
+          } catch (error) {
+            failed.push(email);
+            console.error("Error adding user to organization", error);
+          }
+        }
+
+        // Only show success notification if any invites succeeded
+        if (succeeded.length > 0) {
```

Comment:
We need a case where both `succeeded.length > 0` and `failed.length > 0` are true. Otherwise there will still be overlapping notifications.

Reply:
https://github.com/rilldata/rill/pull/6658/files#r1954936512

Reply:
Made the change directly in NotificationCenter since there shouldn't ever be overlapping notifications in general when it's not supported